tuf/api/metadata.py

```diff
@@ -42,25 +42,17 @@ class Metadata:
         signed: A subclass of Signed, which has the actual metadata payload,
             i.e. one of Targets, Snapshot, Timestamp or Root.
 
-        signatures: A list of signatures over the canonical representation of
-            the value of the signed attribute::
-
-            [
-                {
-                    'keyid': '<SIGNING KEY KEYID>',
-                    'sig':' '<SIGNATURE HEX REPRESENTATION>'
-                },
-                ...
-            ]
+        signatures: A list of Securesystemslib Signature objects, each signing
+            the canonical serialized representation of 'signed'.
 
     """
 
-    def __init__(self, signed: "Signed", signatures: list) -> None:
+    def __init__(self, signed: "Signed", signatures: List[Signature]) -> None:
```

Comment:
I agree we should start using `List` instead of `list` in type annotations mostly because we can annotate what's on the list too. I have been thinking to make that change for a while!